updater/aws.go

```diff
@@ -119,27 +153,42 @@ func (u *updater) filterAvailableUpdates(bottlerocketInstances []instance) ([]in
 		instances = append(instances, inst.instanceID)
 	}
 
-	commandID, err := u.sendCommand(instances, u.checkDocument)
-	if err != nil {
-		return nil, err
-	}
-
+	var lastErr error
+	errCount := 0
 	candidates := make([]instance, 0)
-	for _, inst := range bottlerocketInstances {
-		commandOutput, err := u.getCommandResult(commandID, inst.instanceID)
+	pageCount, err := eachPage(len(instances), ssmPageSize, func(start, stop int) error {
+		commandID, err := u.sendCommand(instances[start:stop], u.checkDocument)
 		if err != nil {
-			return nil, err
-		}
-		output, err := parseCommandOutput(commandOutput)
-		if err != nil {
-			// not a fatal error, we can continue checking other instances.
-			log.Printf("Failed to parse command output %q: %v", string(commandOutput), err)
-			continue
+			// errors here are considered non-fatal.
+			log.Printf("Unable to send document %s: %v", u.checkDocument, err)
```

Comment:
nit: Keeping consistent
```suggestion
			log.Printf("Failed to send document %s: %v", u.checkDocument, err)
```